libcudacxx/include/cuda/std/__bit/reference.h

```diff
@@ -1142,56 +1051,58 @@ bool __equal_aligned(
 equal(__bit_iterator<_Cp, _IC1> __first1, __bit_iterator<_Cp, _IC1> __last1, __bit_iterator<_Cp, _IC2> __first2)
 {
   if (__first1.__ctz_ == __first2.__ctz_)
   {
-    return __equal_aligned(__first1, __last1, __first2);
+    return _CUDA_VSTD::__equal_aligned(__first1, __last1, __first2);
   }
-  return __equal_unaligned(__first1, __last1, __first2);
+  return _CUDA_VSTD::__equal_unaligned(__first1, __last1, __first2);
 }
 
-template <class _Cp, bool _IsConst, typename _Cp::__storage_type>
+template <class _Cp, bool _IsConst>
 class __bit_iterator
 {
 public:
-  typedef typename _Cp::difference_type difference_type;
-  typedef bool value_type;
-  typedef __bit_iterator pointer;
-  typedef typename conditional<_IsConst, __bit_const_reference<_Cp>, __bit_reference<_Cp>>::type reference;
-  typedef random_access_iterator_tag iterator_category;
+  using difference_type   = typename _Cp::difference_type;
+  using value_type        = bool;
+  using pointer           = __bit_iterator;
+  using reference         = __conditional_t<_IsConst, __bit_const_reference<_Cp>, __bit_reference<_Cp>>;
+  using iterator_category = random_access_iterator_tag;
 
 private:
-  typedef typename _Cp::__storage_type __storage_type;
-  typedef typename conditional<_IsConst, typename _Cp::__const_storage_pointer, typename _Cp::__storage_pointer>::type
-    __storage_pointer;
+  using __storage_type = typename _Cp::__storage_type;
+  using __storage_pointer =
+    __conditional_t<_IsConst, typename _Cp::__const_storage_pointer, typename _Cp::__storage_pointer>;
+
   static const unsigned __bits_per_word = _Cp::__bits_per_word;
 
   __storage_pointer __seg_;
   unsigned __ctz_;
 
 public:
-  _LIBCUDACXX_INLINE_VISIBILITY __bit_iterator() noexcept
-#if _CCCL_STD_VER > 2011
+  _LIBCUDACXX_HIDE_FROM_ABI _LIBCUDACXX_INLINE_VISIBILITY _CCCL_CONSTEXPR_CXX14 __bit_iterator() noexcept
       : __seg_(nullptr)
       , __ctz_(0)
-#endif
   {}
-  // avoid re-declaring a copy constructor for the non-const version.
-  using __type_for_copy_to_const = _If<_IsConst, __bit_iterator<_Cp, false>, struct __private_nat>;
 
-  _LIBCUDACXX_INLINE_VISIBILITY __bit_iterator(const __type_for_copy_to_const& __it) noexcept
+  _CCCL_CONSTEXPR_CXX14 __bit_iterator(const __bit_iterator<_Cp, _IsConst>& __it) = default;
+
+  template <bool _OtherIsConst, class = __enable_if_t<_IsConst == true && _OtherIsConst == false>>
+  _LIBCUDACXX_HIDE_FROM_ABI _LIBCUDACXX_INLINE_VISIBILITY _CCCL_CONSTEXPR_CXX14
+  __bit_iterator(const __bit_iterator<_Cp, _OtherIsConst>& __it) noexcept
       : __seg_(__it.__seg_)
       , __ctz_(__it.__ctz_)
   {}
 
-  _LIBCUDACXX_INLINE_VISIBILITY reference operator*() const noexcept
+  _LIBCUDACXX_HIDE_FROM_ABI _LIBCUDACXX_INLINE_VISIBILITY _CCCL_CONSTEXPR_CXX14 reference operator*() const noexcept
   {
-    return reference(__seg_, __storage_type(1) << __ctz_);
+    return __conditional_t<_IsConst, __bit_const_reference<_Cp>, __bit_reference<_Cp>>(
```

Comment:
Could this use the `reference` alias?
```suggestion
    return reference(
```